content/docs/io/reading_writing_data.md

```diff
@@ -6,7 +6,156 @@ menu:
     parent: io
 ---
 
-This page has not been worked on yet. If you'd like to contribute visit the [doc-push]
-repo.
 
-[doc-push]: https://github.com/tokio-rs/doc-push
+
+# Non-blocking I/O
+
+In [the overview] we mentioned briefly that Tokio's I/O types implement
+non-blocking variants of `std::io::Read` and `std::io::Write` called
+[`AsyncRead`] and [`AsyncWrite`]. These are an integral part of Tokio's
+I/O story, and are important to understand when working with I/O code.
+
+> Note: in this section, we'll primarily talk about `AsyncRead`, but
+> `AsyncWrite` is pretty much exactly the same, just for writing data to
+> an I/O resource (like a TCP socket) instead of reading from it.
+
+So, let's take a look at [`AsyncRead`] and see what all the fuzz is
```

Comment:
I believe this idiom is "what all the fuss is about", but maybe this is a variant I'm not familiar with... :-D